packages/renderer/src/lib/feedback/SendFeedback.svelte

```diff
@@ -1,173 +1,24 @@
-        >Tell us why, or share any suggestion or issue to improve your experience: ({1000 - tellUsWhyFeedback.length} characters
-        left)</label>
-      <textarea
-        rows="3"
-        maxlength="1000"
-        name="tellUsWhyFeedback"
-        id="tellUsWhyFeedback"
-        data-testid="tellUsWhyFeedback"
-        bind:value={tellUsWhyFeedback}
-        class="w-full p-2 outline-none text-sm bg-[var(--pd-input-field-focused-bg)] rounded-sm text-[var(--pd-input-field-focused-text)] placeholder-[var(--pd-input-field-placeholder-text)]"
-        placeholder="Please enter your feedback here, we appreciate and review all comments"></textarea>
-
-      <label for="contactInformation" class="block mt-4 mb-2 text-sm font-medium text-[var(--pd-modal-text)]"
-        >Share your contact information if you'd like us to answer you:</label>
-      <input
-        type="email"
-        name="contactInformation"
-        id="contactInformation"
-        bind:value={contactInformation}
-        placeholder="Enter email address, or leave blank for anonymous feedback"
-        class="w-full p-2 outline-none text-sm bg-[var(--pd-input-field-focused-bg)] rounded-sm text-[var(--pd-input-field-focused-text)] placeholder-[var(--pd-input-field-placeholder-text)]" />
-    </svelte:fragment>
-    <svelte:fragment slot="validation">
-      {#if smileyRating === 0}
-        <ErrorMessage class="text-xs" error="Please select an experience smiley" />
-      {:else if smileyRating === 1 && !hasFeedback}
-        <ErrorMessage class="text-xs" error="Please share contact info or details on how we can improve" />
-      {:else if smileyRating === 2 && !hasFeedback}
-        <WarningMessage class="text-xs" error="We would really appreciate knowing how we can improve" />
-      {:else if smileyRating > 2}
-        <div class="text-[var(--pd-modal-text)] p-1 flex flex-row items-center text-xs">
-          <Fa size="1.125x" class="cursor-pointer" icon={faQuestionCircle} />
-          <span aria-label="Like Podman Desktop? Give us a star on GitHub" class="flex items-center">
-            <Fa class="px-1 text-[var(--pd-invert-content-info-icon)]" icon={faHeart} />{smileyRating === 3 ? 'Like' : 'Love'} It? Give us a <Fa
-              class="px-1 text-[var(--pd-state-warning)]"
-              icon={faStar} />on <Link aria-label="GitHub" onclick={openGitHub}>GitHub</Link>
-          </span>
-        </div>
-      {/if}
-    </svelte:fragment>
-    <svelte:fragment slot="buttons">
-      <Button disabled={smileyRating === 0 || (smileyRating === 1 && !hasFeedback)} on:click={() => sendFeedback()}
-        >Send feedback</Button>
-    </svelte:fragment>
-  </Dialog>
+  {#if category === 'developers'}
+    <DevelopersFeedback bind:displayModal={displayModal} bind:category={category} />
+  {:else}
+    <GitHubIssueFeedback bind:displayModal={displayModal} bind:category={category} />
+  {/if}
```

Comment:
IMHO, the Dropdown should be in the `SendFeedback`, and the GitHubIssueFeedback, and DevelopersFeedback  should only be forms that submit to the parent component.

Here we are adding a lot of complexity by binding values to children.

Reply:
since Modal uses slots, is there a way I can connect buttons from the developer feedback or github feedback to the Dialog in SendFeedback?

Reply:
Yes, this might be a little complex and use the svelte 5 reactivity. I wrote the following quickly, to illustrate (do not take as it is)

Let's have a `FeedBackModal`, I like the idea of using our Modal, but we need to make sure this do not introduce any visual regression (maybe it is smaller in width or height?)

```svelte
<script lang="ts">
import type { Component } from 'svelte';
import type { FeedbackFormProps, FormContent } from './feedback-form-props';
import type { FeedbackCategory } from '/@api/feedback';
import GitHubForm from '/@/lib/feedback/GitHubForm.svelte';

// let's have our FeedbackCategory in a state
let category: FeedbackCategory = $state<FeedbackCategory>('developers');
// The content of our form should also be in a state and handled by the parent
let form: FormContent = $state({
  title: '',
  description: '',
});

// We can create a FormComponent state, which 
let FormComponent: Component<FeedbackFormProps> = $derived.by(() => {
  switch (category) {
    case 'developers':
      return DeveloperForm;
    case 'feature':
      return GitHubForm; // might need something dedicated ?
    case 'bug':
      return GitHubForm; // might need something dedicated ? 
  }
});

// function that would handle form change in child component
async function onFormChange(value: FormContent): Promise<void> {
  form = value;
}

// function that would hande dropdown change in this component
function onDropdownChange(nCategory: FeedbackCategory): void {
  // apply new category
  category = nCategory;
  // reset forms when switching category?
  form = {
    title: '',
    description: '',
  };
}
</script>

{/* todo: should handle the hidden stuff */}
<Modal {...}>
  {/* Dropdown */}
  <FormComponent category={category} onchange={onChange} />
  {/* Buttons */}
</Modal>
```

Here are some types I wrote for my example (dedicated file so several component can import and share it)

```ts
import type { FeedbackCategory } from '/@api/feedback';

export interface FormContent {
  // mandatory properties
  title: string;
  description: string;
}

// This should be the props of our form components, it should be shared
export interface FeedbackFormProps {
  category: FeedbackCategory;
  value: FormContent;
  onchange: (value: FormContent) => Promise<void>;
}
```

Let's see how our GitHubForm component would look like, the idea here, would to only have the forms field, the child component will update the value of the form using the onchange props provided

```svelte
<script lang="ts">
import type { FeedbackFormProps } from '/@/lib/feedback/feedback-form-props';
import { Input } from '@podman-desktop/ui-svelte';

let { category, onchange, value }: FeedbackFormProps = $props();

function onTitleChange(e: Event) {
  onchange({
    ...value,
    title: e.event.target,
  });
}

function onDescriptionChange(e: Event) {
  onchange({
    ...value,
    description: e.event.target,
  });
}
</script>

{/* todo: handle ui nicely with labels, styling etc */}
<Input value={value.title} title="title" on:change={onTitleChange}></Input>

{/* todo: handle ui nicely with labels, styling etc */}
<Input value={value.description} title="description" on:change={onDescriptionChange}></Input>
```